beacon-chain/powchain/engine_client.go

```diff
@@ -65,9 +69,11 @@ func (s *Service) NewPayload(ctx context.Context, payload *pb.ExecutionPayload)
 	defer func() {
 		newPayloadLatency.Observe(float64(time.Since(start).Milliseconds()))
 	}()
-
+	d := time.Now().Add(payloadAndForkchoiceUpdatedTimeout)
+	ctx, cancel := context.WithDeadline(ctx, d)
 	result := &pb.PayloadStatus{}
 	err := s.rpcClient.CallContext(ctx, result, NewPayloadMethod, payload)
+	cancel()
```

Comment:
```suggestion
	ctx, cancel := context.WithDeadline(ctx, d)
	defer cancel()
	result := &pb.PayloadStatus{}
	err := s.rpcClient.CallContext(ctx, result, NewPayloadMethod, payload)
```

Usually the pattern is to defer immediately after creating the cancel fn. 
Suggestion is to conform to the normal pattern of deferring cancel here and in all places this was added in the PR